src/Illuminate/Http/Request.php

```diff
@@ -640,6 +640,37 @@ public function accepts($contentTypes)
     }
 
     /**
+     * Determines whether the current requests accepts a given content type
+     * based on content negotiation.
+     *
+     * @param  string|array  $contentTypes
+     * @return string|null
+     */
+    public function wants($contentTypes)
+    {
+        $accepts = $this->getAcceptableContentTypes();
+        $contentTypes = (array) $contentTypes;
+
+        foreach ($accepts as $accept) {
+            if (in_array($accept, ['*/*', '*'])) {
+                return $contentTypes[0];
+            }
+
+            foreach ($contentTypes as $contentType) {
+                $type = $contentType;
+
+                if (!is_null($mimeType = $this->getMimeType($contentType))) {
+                    $type = $mimeType;
+                }
+
+                if ($this->matchesType($type, $accept) || $accept === strtok($type, '/').'/*') {
```

Comment:
Accept  | Wants | Return
|----------|----------|-----------
 application/* | application/json | true
 application/* | text/html | false
 application/* | application/xml | true

> You need to swap the $accept and $type arguments.

It would be extremely odd if you actually want to do `$request->wants('application/*');` as `application/xml` and `application/json` is completely different format.